### ISSUE TITLE
P87 Confusion on static::$rules

### ISSUE THREAD

**Issue by sorora** [JeffreyWay/Laravel-Testing-Decoded#25]
Hello,

On page 87 in Chapter 6: Testing Models
(referencing: <?php // app/tests/BaseModelTest.php)

$model::$rules = ['title' => 'required'];

The above bit of code is shown, okay, no problem ... go down later to where the BaseModel is defined, no $rules property is defined at all, and when it comes to page 89 where the Author model is set up to extend the BaseModel, the $rules is public, but then in next two examples change to protected.

I have no idea if this is right or wrong, I'm trying to follow along while testing a package, but when I try and do "$model::$rules" to test the BaseModel, it says "fatal error: access to undeclared static property".

TL;DR = Not sure if I'm in wrong or book has a typo essentially.

**Comment by rand0mb1ts** [comment]
I also ran into this problem as well.  

BaseModel needs 'pulblic static $rules=[ ]' to pass testing that is given in the book.

Then in the child models you must use 'public static $rules = [  ... ]' to define your rules because $rules is public in BaseModel.

**Comment by rtablada** [comment]
You should be good to declare just

``` php
static $rules;
```

That is what I did and it worked up to the BaseModel test. I think by declaring no accessibility, you should be able to extend and make it protected. Will test that as I am just about to do the AuthorTest.

**Comment by sorora** [comment]
Declaring no accessibility = public by default.

**Comment by rtablada** [comment]
Been in PHP for a while but somethings like that sometimes get stuffed to the back of the mental shelf with all the languages that I've learned.

**Comment by JeffreyWay** [comment]
Hey, guys. Sorry - I've fixed that part of the book and have republished.

**Comment by joustava** [comment]
You sure you fixed it?
Bought it yesterday. This version was published on 2013-06-08.

I still have an issue. Yes, when I add the $rules line to the BaseModel the tests do not fail anymore. It seems however that they are false positives. When I change the test to test for a valid record it fails (different strings for names).

``` php
public function testIsInValidWithoutUniqueName()
  {
    $chore = Factory::create('chore', ['name' => 'Dodishes']);
    $chore = Factory::chore(['name' => 'Do dishes']);

    $this->assertValid($chore);      
  }
```

The rest of my code is [here](https://github.com/joustava/points/tree/ltd). It is slightly modified code.

**Comment by rand0mb1ts** [comment]
I had to make sure my model names matches exactly... try using Chore instead of chore.

**Comment by joustava** [comment]
Tried that. Still the same result.

**Comment by rand0mb1ts** [comment]
Have you migrated and seeded the database?

**Comment by joustava** [comment]
Yes, I am using Factories and clean the db before each test. I seem to do everything similar to the examples given.
This should be valid.

``` php
 public function testIsInValidWithoutUniqueName()
  {
    $chore = Factory::create('Chore', ['name' => 'joost']);
    $chore = Factory::Chore(['name' => 'other']);

    $this->assertValid($chore);      
  }
```

**Comment by hervejodoin** [comment]
Purchased book 2 weeks ago. Error still present.

**Comment by mburakergenc** [comment]
public  $rules = [

```
'username' => 'required',
'password' => 'required'
```

];

Just make it public "static" $rules. Then you will be good to go.
